pythonFiles/vscode_pytest/__init__.py

```diff
@@ -0,0 +1,227 @@
+            if test_class_node not in test_file_node["children"]:
+                test_file_node["children"].append(test_class_node)
+    created_files_folders_dict: dict[str, TestNode] = {}
+    for file_module, file_node in file_nodes_dict.items():
+        # Iterate through all the files that exist and construct them into nested folders.
+        root_folder_node: TestNode = build_nested_folders(
+            file_module, file_node, created_files_folders_dict, session
+        )
+        # The final folder we get to is the highest folder in the path and therefore we add this as a child to the session.
+        if root_folder_node.get("id_") not in session_children_dict:
+            session_children_dict[root_folder_node.get("id_")] = root_folder_node
+    session_node["children"] = list(session_children_dict.values())
+    return session_node, errors
+
+
+def build_nested_folders(
+    file_module: pytest.Module,
+    file_node: TestNode,
+    created_files_folders_dict: dict[str, TestNode],
+    session: pytest.Session,
+) -> TestNode:
+    prev_folder_node: TestNode = file_node
+
+    # Begin the i_path iteration one level above the current file.
+    iterator_path: pathlib.Path = file_module.path.parent
+    while iterator_path != session.path:
+        curr_folder_name: str = iterator_path.name
+        try:
+            curr_folder_node: TestNode = created_files_folders_dict[curr_folder_name]
+        except KeyError:
+            curr_folder_node: TestNode = create_folder_node(
+                curr_folder_name, iterator_path
+            )
+            created_files_folders_dict[curr_folder_name] = curr_folder_node
+        if prev_folder_node not in curr_folder_node["children"]:
+            curr_folder_node["children"].append(prev_folder_node)
+        iterator_path = iterator_path.parent
+        prev_folder_node = curr_folder_node
+    return prev_folder_node
+
+
+def create_test_node(
+    test_case: pytest.Item,
+) -> TestItem:
+    test_case_loc: str = (
+        "" if test_case.location[1] is None else str(test_case.location[1] + 1)
+    )
+    return {
+        "name": test_case.name,
+        "path": str(test_case.path),
+        "lineno": test_case_loc,
+        "type_": TestNodeTypeEnum.test,
+        "id_": test_case.nodeid,
+        "runID": test_case.nodeid,
+    }
+
+
+def create_session_node(session: pytest.Session) -> TestNode:
+    return {
+        "name": session.name,
+        "path": str(session.path),
+        "type_": TestNodeTypeEnum.folder,
+        "children": [],
+        "id_": str(session.path),
+    }
+
+
+def create_class_node(class_module: pytest.Class) -> TestNode:
+    return {
+        "name": class_module.name,
+        "path": str(class_module.path),
+        "type_": TestNodeTypeEnum.class_,
+        "children": [],
+        "id_": class_module.nodeid,
+    }
+
+
+def create_file_node(file_module: pytest.Module) -> TestNode:
+    return {
+        "name": str(file_module.path.name),
```

Comment:
```suggestion
        "name": str(file_module.path.name),
```

Can this be anything but a string?